components/style/properties/properties.mako.rs

```diff
@@ -1572,6 +1599,28 @@ pub struct VariableDeclaration {
     value: Arc<UnparsedValue>,
 }
 
+impl ToCss for VariableDeclaration {
+    fn to_css<W>(&self, dest: &mut CssWriter<W>) -> fmt::Result
+    where
+        W: fmt::Write,
+    {
+        // https://drafts.csswg.org/css-variables/#variables-in-shorthands
+        match self.value.from_shorthand {
+            // Normally, we shouldn't be printing variables here if they came from
+            // shorthands. But we should allow properties that came from shorthand
+            // aliases. That also matches with the Gecko behavior.
```

Comment:
Can you mention here that this is just a hack for `-moz-transform`, which is really a shorthand for who knows what reason?

Also we should try to unship that, so feel free to also add a `FIXME(emilio)` over there so I eventually do it.